client/src/App.js

```diff
@@ -1,29 +1,39 @@
 import React, { useState, useEffect } from 'react';
-import { BrowserRouter, Route } from 'react-router-dom';
+import { BrowserRouter } from 'react-router-dom';
+import { CssBaseline } from '@material-ui/core';
+import { useDispatch } from 'react-redux';
+import Pages from './pages/Pages';
+import { setUser} from './store/auth';
+ 
 function App() {
   const [loading, setLoading] = useState(true);
+  const dispatch = useDispatch();
 
   useEffect(() => {
     const loadUser = async () => {
       // enter your back end route to get the current user
       const res = await fetch("/api/session");
       if (res.ok) {
         res.data = await res.json(); // current user info
+        dispatch(setUser(res.data.user))
       }
       setLoading(false);
     }
+
     loadUser();
-  }, []);
+  }, [dispatch]);
```

Comment:
You can actually remove `dispatch` from the dependency array. Check out this [StackOverflow](https://stackoverflow.com/questions/54930197/react-hooks-dispatch-action-from-useeffect) and note that someone said this:
> React guarantees that dispatch function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.